server/hydrogen-render/2-render-hydrogen-to-string-fork-script.js

```diff
@@ -6,14 +6,60 @@
 
 const assert = require('assert');
 
+const RethrownError = require('../lib/rethrown-error');
 const _renderHydrogenToStringUnsafe = require('./3-render-hydrogen-to-string-unsafe');
 
+// Serialize the error and send it back up to the parent process so we can
+// interact with it and know what happened when the process exits.
+async function serializeError(err) {
+  await new Promise((resolve) => {
+    process.send(
+      {
+        error: true,
+        name: err.name,
+        message: err.message,
+        stack: err.stack,
+      },
+      (sendErr) => {
+        if (sendErr) {
+          // We just log here instead of rejecting because it's more important
+          // to see the original error we are trying to send up. Let's just
+          // throw the original error below.
+          const sendErrWithDescription = new RethrownError(
+            'Failed to send error to the parent process',
+            sendErr
+          );
+          console.error(sendErrWithDescription);
+          // This will end up hitting the `unhandledRejection` handler and
+          // serializing this error instead (worth a shot) 🤷‍♀️
+          throw sendErrWithDescription;
+        }
+
+        resolve();
+      }
+    );
+  });
+}
+
+// We don't exit the process after encountering one of these because maybe it
+// doesn't matter to the main render process in Hydrogen.
+//
+// If we don't listen for these events, the child will exit with status code 1
+// (error) when they occur.
+process.on('uncaughtException', async (err /*, origin*/) => {
+  await serializeError(new RethrownError('uncaughtException in child process', err));
+});
```

Comment:
The really nice fix in this PR is listening to `uncaughtException` here ^

We keep track of the errors to display if we fail to get output from Hydrogen so it's easier to debug.

But these don't fail the process because they're out of band from the main Hydrogen rendering process and don't necessarily mean the Hydrogen render failed. Like `localStorage` and `IndexDB` errors.